aiida_quantumespresso/cli/calculations/dos.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+"""Command line scripts to launch a `PwCalculation` for testing and demonstration purposes."""
+from __future__ import absolute_import
+import click
+
+from aiida.orm import load_node
+from aiida.orm import RemoteData, CalcJobNode
+from aiida.cmdline.params import options, types
+from aiida.cmdline.utils import decorators
+
+from aiida_quantumespresso.cli.utils import options as options_qe
+from aiida_quantumespresso.cli.utils import validate
+
+# required inputs: code, input_parameters, (parent_calculation or parent_folder)
+# optional: settings
+
+@click.command()
+@options.CODE(required=True, type=types.CodeParamType(entry_point='quantumespresso.dos'))
+#@options.CALCULATION(required=True, type=types.CalculationParamType(sub_classes=('aiida.calculations:quantumespresso.pw',)))
+#@options.CALCULATION(required=True, type=CalcJobNode)
+@options.CALCULATION(required=True)
```

Comment:
@sphuber do you have a suggestion on how to do ask for a calculation from a specific plugin? It seems that now all `CalcJobNode`s are equal, i.e. there's no such thing as a `PwCalculationNode`, and therefore I cannot ask for a specific class either directly or via the entry point system. Is this correct?

Reply:
Passing `sub_classes=('aiida.calculations:quantumespresso.pw',)` tells me that `PwCalculation` is not a subclass of `CalculationNode`.

Reply:
Yes this is correct, because `PwCalculation` is a sub class of `Process` and not of `Node`. To be able to filter for nodes that were created by the `PwCalculation` process class, you would have to filter on the `process_type` of the nodes. This is currently not implemented in the `IdentifierParamType`.

Reply:
Ok, thanks a lot.